### ISSUE TITLE
Remotely service failed to start due to the following error

### ISSUE THREAD

**Issue by swilkey** [Jay-Rad/Remotely#82]
Hi Jared,

I've just seen two machines with this problem.  One was installed yesterday and I successfully remote controlled it last night from home.  The other was just installed a few minutes ago.  Both were showing green on the grid.  When I tried to remote control them just now it appeared to connect but I never saw their screens.  When we then checked the remotely service on those machines we found it was not running.  When we tried to start the remotely service on those machines we got this error:
Remotely service failed to start due to the following error: The system cannot find the file specified.

Windows Event Viewer shows this in the System log, as coming from the Service Control Manager.  It is event id 7000.  There is no other information logged and therefore no way of knowing what the missing file is.

However, I looked in services and found that this is the file that is supposed to be run "C:\Program Files\Remotely\Remotely_Agent.exe".  We checked the machine that had only just been installed with remotely and discovered that "C:\Program Files\Remotely\" is an empty folder.  I don't know why this would be.  Have you come across it before?

Unfortunately I couldn't check the other computer that had this problem as I tried reinstalling remotely on it and it worked fine.  I wondered if this might have been something due to a false virus detection in the tool, but windows defender hasn't detected any issues.

Having now reinstalled remotely on both machines I find I cannot actually remote control them or chat with them even though they show up green in the grid.  I looked in the c:\program files\remotely folder on one of them and the files are there.  I left that explorer window open and now, 5 minutes later I can see that all the files are now MISSING from that folder!  What could be happening here?  Is there some sort of automatic uninstallation process that happens if an agent connection fails?  I'll post again if I discover anything new.

**Comment by swilkey** [comment]
I enabled auditing on the c:\program files\remotely folder on the computer that had uninstalled itself twice so I could try to understand what was happening, but since then it has not disappeared and remote control is working.

I do notice that chat is not working with any devices and I don't understand why.  I cannot find any logging of attempted chat sessions (or even successful ones from yesterday for that matter) so I don't get it.  The little chat window opens on top of the grid, but never does anything further (as would happen if a chat were successful).

**Comment by swilkey** [comment]
I just discovered the documentation about the location of the log files on the client. Looking at the log I see this entry at about the time that it disappeared: Service Updater: Update found. Current Version: 2020.03.08.2141.  Latest Version: 2020.03.08.1938; waiting 5 seconds before updating; service updater: downloading installer.  The next entry relates to starting a new screencasting session and makes me think it is after I reinstalled the client again.  

The same thing happened later in the day, at about the time it disappeared the second time.  My question is, where is it obtaining this new version from?  Is there a way I can prevent this behaviour?  Is it possible that the new installation process is failing or is it just that I didn't realise what it was doing and reinstalled before it had finished?
thanks!

**Comment by Jay-Rad** [comment]
The agent auto-updates when the version changes on the server.  It looks like the CurrentVersion.txt file on your server is changing, which is triggering an auto-update.  Nothing in Remotely changes that file, so something on your server is.

There's also a fail-safe that will cause the agent to uninstall itself if tripped.  It sounds like that might be happening as well.

With all the things you described, it sounds like something's messed up with your server.  It seems like data is changing when it shouldn't, and websocket sessions aren't being properly persisted.

Could you try creating a server on Vultr or Digital Ocean?  A $5/month server will run Remotely just fine.  And I've verified that it works on those providers.

**Comment by Jay-Rad** [comment]
I should also clarify, the Ubuntu 18.04  or 19.10 image.  Those are the ones I've used.  :)

**Comment by swilkey** [comment]
Hi Jared,

Thanks for your reply.  I am using Ubuntu 18.04.  I think I understand what happened.  I installed the server using the installation script you provided.  Then I realised that the agent was not preloaded with the name of my server, so I copied the agent and instant support clients from the version of the server that I built on my computer as I reasoned that they would have the correct information in them.  However, they obviously were built at different times and therefore had different build information.  Not knowing about the CurrentVersion.txt file I never updated that, so it obviously had the original details in it and was continually wanting to update any clients because every time it updated them they still had the wrong version information.  Thanks for solving this for me!
